### PR TITLE
Opt-in for using hipComplex types

### PR DESCRIPTION
**Pull request by daineAMD**
See https://github.com/ROCmSoftwarePlatform/rocBLAS-internal/pull/863. The sample code is a direct copy from there.

### CODE REVIEW

File: library/include/hipblas.h
Comment (TorreZuk) [inline library/include/hipblas.h:216]: Does it fail?   It must have passed before no?    The sizeof being 2 * floats etc.  may be equivalent ...

Comment (daineAMD) [reply library/include/hipblas.h:216]: It fails with hipComplex types. I'm not sure why we need the check myself [cpp reference](https://en.cppreference.com/w/cpp/types/is_trivial)

Comment (TorreZuk) [reply library/include/hipblas.h:216]: I think you don't want any default constructor doing anything different than C, for us we are covered by the C++ version  based definitions so maybe just move it into the #else our internal type checking.   Those who opt-into hip complex get what they asked for and the size compatibility is all we want to verify and standard layout.

**Review by TorreZuk — APPROVED** [review]
LGTM if all tests pass, but we need to wait for general approval.

**Review by evetsso — APPROVED** [review]
I have no objections.  rocFFT uses void*'s in its public API.  hipFFT already uses hipComplex and hipDoubleComplex, and does not expose any custom types.

**Pull request merged**
